### ISSUE TITLE
Cache option doesn't work well with use of downloaded binaries

### ISSUE THREAD

**Issue by Antoine4011** [Carthage/Carthage#1892]
* `carthage version`: 0.20.1
* `xcodebuild -version`: Xcode 8.3.2 Build version 8E2002
* Are you using `--no-build`? no
* Are you using `--no-use-binaries`? no
* Are you using `--use-submodules`? no
* Are you using `--cache-builds`? yes

**Cartfile**
```
github "ReactiveX/RxSwift"
github "RxSwiftCommunity/RxDataSources"
```

**Carthage Output**
```
*** Fetching RxDataSources
*** Fetching RxSwift
*** Downloading RxSwift.framework binary at "David"
*** Checking out RxDataSources at "1.0.3"
*** xcodebuild output can be found in /var/folders/7y/2yx1lbwd56x60hsfrmqr79l80000gn/T/carthage-xcodebuild.TRZolb.log
*** No cache found for RxSwift, building with all downstream dependencies
*** Building scheme "RxDataSources-iOS" in RxDataSources.xcworkspace
```

The RxSwift repo has a valid binary, so Carthage downloads it. (So far it's great). 
But I have this message saying that the cache for RxSwift isn't found. (However Carthage will still use the downloaded binary instead of building with all downstream dependencies)

But RxSwift being a dependency for RxDataSources-iOS, it somehow prevents the use of the RxDataSources-iOS's cache and builds it everytime.

In order to make the cache work, I would need to use `--no-use-binaries`.

**Comment by edfanhouse** [comment]
We have the same issue here with Realm:

*** Downloading realm-cocoa.framework binary at "v2.6.2"
*** No cache found for realm-cocoa, building with all downstream dependencies
*** Building scheme "Realm" in Realm.xcworkspace

**Comment by mdiep** [comment]
This is being addressed in #1850.

**Comment by erichoracek** [comment]
This appears to have been fixed in 0.21, can you confirm @Antoine4011?

```
$ pbpaste > Cartfile

$ carthage bootstrap
*** No Cartfile.resolved found, updating dependencies
*** Cloning RxDataSources
*** Cloning RxSwift
*** Downloading RxSwift.framework binary at "David"
*** Checking out RxDataSources at "1.0.3"
*** xcodebuild output can be found in /var/folders/9s/rl0z7g591cs9frc53y6cbpzm0000gq/T/carthage-xcodebuild.jN2u9p.log
*** Building scheme "RxDataSources-tvOS" in RxDataSources.xcworkspace
*** Building scheme "RxDataSources-iOS" in RxDataSources.xcworkspace

$ carthage build --cache-builds
*** xcodebuild output can be found in /var/folders/9s/rl0z7g591cs9frc53y6cbpzm0000gq/T/carthage-xcodebuild.bo8gVg.log
*** Valid cache found for RxSwift, skipping build
*** Valid cache found for RxDataSources, skipping build
```
